src/main/java/com/opinta/controller/ShipmentController.java

```diff
@@ -51,9 +51,10 @@ public ShipmentController(ShipmentService shipmentService, PDFGeneratorService p
         return new ResponseEntity<>(shipmentDto, OK);
     }
 
-    @GetMapping("{id}/label-form")
-    public ResponseEntity<?> getShipmentLabelForm(@PathVariable("id") long id) {
-        byte[] data = pdfGeneratorService.generateLabel(id);
+    @GetMapping("{id}/{parcelId}/label-form")
+    public ResponseEntity<?> getShipmentLabelForm(@PathVariable("id") long id,
+                                                  @PathVariable("parcelId") long parcelId) {
+        byte[] data = pdfGeneratorService.generateLabel(id, parcelId);
```

Comment:
Not sure what to do in this case . Since field from shipment and parcel where separated and both are used by pdf gen. So there is a need to know what parcel from collection to use for pdf generation.